sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,898 @@
+  DigitalTwinsGetByIdResponse,
+  DigitalTwinsAddResponse,
+  DigitalTwinsUpdateOptionalParams,
+  DigitalTwinsUpdateResponse,
+  DigitalTwinsDeleteOptionalParams,
+  DigitalTwinsGetComponentResponse,
+  DigitalTwinsUpdateComponentResponse,
+  DigitalTwinsUpdateComponentOptionalParams,
+  DigitalTwinsAddRelationshipResponse,
+  DigitalTwinsAddRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipResponse,
+  DigitalTwinsDeleteRelationshipOptionalParams,
+  DigitalTwinsSendTelemetryOptionalParams,
+  DigitalTwinsSendComponentTelemetryOptionalParams,
+  DigitalTwinsListRelationshipsResponse,
+  IncomingRelationship,
+  DigitalTwinsListIncomingRelationshipsResponse,
+  ModelData,
+  DigitalTwinModelsGetByIdResponse,
+  DigitalTwinModelsGetByIdOptionalParams,
+  DigitalTwinModelsAddResponse,
+  DigitalTwinModelsAddOptionalParams,
+  DigitalTwinModelsListResponse,
+  DigitalTwinModelsListOptionalParams,
+  DigitalTwinModelsListOptions,
+  EventRoutesGetByIdResponse,
+  EventRoute,
+  EventRoutesAddOptionalParams,
+  EventRoutesListNextResponse,
+  EventRoutesListOptionalParams,
+  QueryQueryTwinsResponse,
+  QuerySpecification
+} from "./generated/models";
+
+/**
+ * Client for Azure IoT DigitalTwins API.
+ */
+export class DigitalTwinsClient {
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated AzureDigitalTwinsAPI
+   */
+  private readonly client: GeneratedClient;
+
+  // private readonly client!: AzureDigitalTwinsAPI;
+
+  /**
+   * Creates an instance of AzureDigitalTwinsAPI.
+   *
+   * Example usage:
+   * ```ts
+   * const { DigitalTwinsClient, ServiceClientCredentials } = require("@azure/digitaltwins");
+   *
+   * const client = new DigitalTwinsClient(
+   *   "<endpoint>",
+   *   new DefaultAzureCredential();
+   * );
+   * ```
+   * @param {string} endpoint The endpoint of the service. If options parameter is given than this value will be ignored.
+   * @param {TokenCredential} credential Used to authenticate requests to the service.
+   * @param {AzureDigitalTwinsAPI} [options] Used to configure the service client.
+   */
+  constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options: AzureDigitalTwinsAPIOptionalParams = {}
+  ) {
+    const authPolicy = bearerTokenAuthenticationPolicy(
+      credential,
+      "https://digitaltwins.azure.net/.default"
+    );
+    this.client = new GeneratedClient(credential, {
+      ...{ endpoint },
+      ...createPipelineFromOptions({}, authPolicy),
+      ...options
+    });
+  }
+
+  /**
+   * Get a digital twin
+   *
+   * @param digitalTwinId The Id of the digital twin.
+   * @param options The operation options
+   * @returns The application/json digital twin and the http response.
+   */
+  public getDigitalTwin(
+    digitalTwinId: string,
+    options: OperationOptions = {}
+  ): Promise<DigitalTwinsGetByIdResponse> {
+    return this.client.digitalTwins.getById(digitalTwinId, options);
```

Comment:
you'll want to wrap each public facing operation method in an OpenTelemetry span:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts#L256